client/src/file/mod.rs

```diff
@@ -475,6 +524,29 @@ impl Keyring {
 
         self.write().await
     }
+
+    /// Delete any item that cannot be decrypted with the key associated to the
+    /// keyring.
+    ///
+    /// This can only happen if an item was created using
+    /// [`Self::load_unchecked`] or prior to 0.3 where we didn't validate
+    /// the secret when using [`Self::load`] or modified externally.
+    pub async fn delete_broken_items(&self) -> Result<usize, Error> {
+        let key = self.derive_key().await?;
+        let mut keyring = self.keyring.write().await;
+        let mut broken_items = vec![];
+        for (index, encrypted_item) in keyring.items.iter().enumerate() {
+            if let Err(_err) = encrypted_item.clone().decrypt(&key) {
+                broken_items.push(index);
+            }
+        }
+        let n_broken_items = broken_items.len();
+        for index in broken_items {
+            keyring.items.remove(index);
+        }
+        drop(keyring);
+        Ok(n_broken_items)
```

Comment:
Should this call `self.write().await` before returning to persist the changes to the file? It seems like all other methods that make modifications in the Keyring do.